R/helper_lm_robust_fit.R

```diff
@@ -535,8 +530,24 @@ get_fstat <- function(tss_r2s,
   return(f)
 }
 
-compute_fstat <- function(coefs, vcov_fit, nomdf) {
-  crossprod(coefs, chol2inv(chol(vcov_fit)) %*% coefs) / nomdf
+compute_fstat <- function(coef_matrix, coef_indices, vcov_fit, rank, nomdf) {
+  fstat <- tryCatch({
+    vapply(seq_len(ncol(coef_matrix)),
```

Comment:
Not everything needs to be an apply - I would keep the original function around, then use it in a for-loop.

Also you can simplify your sandwich using backsolve.

Reply:
So I could keep the original function, but this is now used in a couple places so it would just be two functions, where one is only called by the other.

In other news, why for over vapply?

Reply:
I think a for-loop would make it clearer what you're doing - 

actually, is there any reason not to factor the block matrix directly, and vectorize the entire calculation, instead of piece by piece? Are there some zeros on the diagonal or something else that would make it fail?